app_flair.py

```diff
@@ -113,129 +134,227 @@ def parse_prediction_request(json_object: JsonDict) -> PredictionRequest:
-                i += 1
-                tokens.append(token)
-            s = Sentence()
-            s.tokens = tokens
-            list_sentences.append(s)
+        cas = load_cas_from_xmi(document["xmi"], typesystem=typesystem)
+        list_from_document = flair_train_ner_dataset(cas)
+        list_sentences.extend(list_from_document)
     train, dev, = train_test_split(list_sentences, train_size=0.8)
     return train, dev
 
 
-def train_ner(train, dev):
+def flair_train_ner_dataset(cas: Cas) -> list:
+    # In flair the instance of Sentence is used to train the model
+    # Each document is transformed into a list of instances of Sentence with the BIO encoding of NER tags
+    # If a sentence in the document has the wrong NER value which is not in the tagset, the sentence will be discarded
+    tagset = [b.decode("utf-8") for b in list(tagger_ner.tag_dictionary.idx2item)]
+    sentence_list = cas.select(SENTENCE_TYPE)
+    list_sentences = []
+    for sentence in sentence_list:
+        incomplete_sentence, tokens = bio_encoding(cas, sentence, tagset, Token)
+        if incomplete_sentence:
+            continue
+        s = Sentence()
+        s.tokens = tokens
+        list_sentences.append(s)
+    global test_sentences
+    test_sentences = list_sentences
+    # unittest.main()
+    return list_sentences
+
+
+def bio_encoding(cas, sentence, tagset, t):
+    i = 0
+    j = 0
+    token_list = list(cas.select_covered(TOKEN_TYPE, sentence))
+    ner_list = list(cas.select_covered(NER_TYPE, sentence))
+    token_list_len = len(token_list)
+    ner_list_len = len(ner_list)
+    tokens = []
+    incomplete_sentence = False
+
+    # In the BIO encoding, "B-tag" is the begin of entity, "I-tag" is the continuation of entity
+    # and "O" is no entity.
+    while i < token_list_len and j < ner_list_len:
+        if ner_list[j].value is None or "B-" + ner_list[j].value not in tagset:
+            incomplete_sentence = True
+            break
+        token = t(cas.get_covered_text(token_list[i]))
+        if token_list[i].begin == ner_list[j].begin:
+            token.add_tag("ner", "B-" + ner_list[j].value)
+            if token_list[i].end == ner_list[j].end:
+                j += 1
+            i += 1
+        elif (
+            token_list[i].begin > ner_list[j].begin
+            and token_list[i].end <= ner_list[j].end
+        ):
+            token.add_tag("ner", "I-" + ner_list[j].value)
+            if token_list[i].end == ner_list[j].end:
+                j += 1
+            i += 1
+        else:
+            token.add_tag("ner", "O")
+            i += 1
+        tokens.append(token)
+    if incomplete_sentence or ner_list_len == 0:
+        return True, []
+    while i < token_list_len:
+        token = Token(cas.get_covered_text(token_list[i]))
+        token.add_tag("ner", "O")
+        i += 1
+        tokens.append(token)
+    return False, tokens
+
+
+def flair_train_ner(train: list, dev: list):
```

Comment:
In python typing, you should use the generic `List` type, see e.g. https://docs.python.org/3/library/typing.html#typing.List